### ISSUE TITLE
How to use BLEU in OpenNMT?

### ISSUE THREAD

**Issue by PosoSAgapo** [OpenNMT/OpenNMT-py#2146]
Hi, I noticed that the BLEU criterion is added in the OpenNMT library. However, as I checked in the documentation, there is no example to show how to use it in the train/valid YAML configuration.  Is the xent criteria the BLEU score? Or what should I do to add BLEU as a criterion?

**Comment by francoishernandez** [comment]
No, BLEU is not embedded here. To get BLEU scores you can translate your evaluation sets and then compute BLEU with regards to your reference with [sacrebleu](https://github.com/mjpost/sacreBLEU).

As for why it's not embedded during training/validation, it's not that straightforward. See #1158 and related issues, as well as our previous discussion in #2129.
